site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+* Click USE PROVIDER and Copy Information
+* Create the `main.tf` Terraform configuration with the copied information
+* Add the previously created key as api_key and add the host as configuration options
+* Make sure host is the full URL including protocol of your FusionAuth instance, in the docker example this may be `http://localhost:9011`
+
+[source]
+----
+terraform {
+  required_providers {
+    fusionauth = {
+      source = "gpsinsight/fusionauth"
+      version = "0.1.94"
+    }
+  }
+}
+
+provider "fusionauth" {
+  api_key = "yqTs2LSMd23DYgwbPBA_AxmXHm7_qNFT9jbi3wL9Jr65ymNkNnFisAvjY"
+  host = "https://your.fusionauth.domain"
+}
+----
+
+Finally, the following command performs several different initialization steps in order to prepare the current working directory for use with Terraform.
+
+[source]
+----
+terraform init
+----
+
+It is always safe to run link:https://developer.hashicorp.com/terraform/cli/commands/init[init] multiple times, to bring the working directory up to date with changes in the configuration. Though subsequent runs may give errors, this command will never delete your existing configuration or state.
+
+=== Configuration Strategy
+
+Once we initialized the Terraform working directory we need to understand the different strategies to handle the configuration which we describe in detail in the following sections.
+
+[cols="2,4,4"]
+|===
+| Strategy | Advantages | Disadvantages
+
+| import | manage existing resources | some resources dont fully support the Terraform lifecycle
+| data source  | reference existing resources without having to manage them | requires a external process to manage these resources
+| create | create and then manage all resources through Terraform | bigger initial effort
+|===
+
+== Terraform Import
+
+Before we go in to Creating, Updating and Removing of resources we first look in to the link:https://developer.hashicorp.com/terraform/language/import[Import] and link:https://developer.hashicorp.com/terraform/language/data-sources[Data Source] functionality. If you have existing resources you want to manage via Terraform you need to import the current configuration to be able to manage it via Terraform.
+
+[NOTE.note]
+====
+The cleaner way is to create all resources through Terraform instead, but in certain scenarios you will have to use `Import` or `Data Source` and we go through a few examples in this and the next section.
+====
+//Additional input for documentation tbd. import cli https://developer.hashicorp.com/terraform/cli/import import config https://developer.hashicorp.com/terraform/language/import https://spacelift.io/blog/importing-exisiting-infrastructure-into-terraform  https://medium.com/swlh/importing-existing-infrastructure-into-terraform-a6ae168ad2bb https://stackoverflow.com/questions/47613926/import-all-resources-defined-in-tf-file https://www.bitslovers.com/terraform-import/)
```

Comment:
depending on successfully solving https://github.com/gpsinsight/terraform-provider-fusionauth/issues/214 the document could be extended with fully importing existing config without having to add anything manually.

As we have this issue I can remove the references from the doc for now.

Reply:
Feel free to link to the GH issue directly in the doc if it would be helpful for users.

Reply:
I'll leave it out, as its mentioned at the top that: Since it is a community project, it is not as complete as the client libraries, but it covers the majority of use cases. If you find yourself needing to manage an unsupported resource or configuration, please submit a PR to the project using GitHub.